src/geqdsk_imas.jl

```diff
@@ -475,22 +475,38 @@ function imas2geqdsk(
         zbbbs = eqt.boundary.outline.z ./ tc["Z"]
         nbbbs = length(rbbbs)
 
-        # 1D profiles
-        p1 = eqt.profiles_1d
-        psi = p1.psi ./ tc["PSI"]
-        qpsi = p1.q ./ tc["Q"]
-        pres = p1.pressure ./ tc["P"]
-        pprime = p1.dpressure_dpsi ./ tc["PPRIME"]
-        fpol = p1.f ./ tc["F"]
-        ffprim = p1.f_df_dpsi ./ tc["F_FPRIME"]
-        rhovn = p1.rho_tor_norm
-
         # 2D flux map
-        p2 = eqt.profiles_2d[1]
+        p2 = findfirst(:rectangular, eqt.profiles_2d)
         r = p2.grid.dim1 ./ tc["R"]
         z = p2.grid.dim2 ./ tc["Z"]
         psirz = p2.psi ./ tc["PSI"]
 
+        # 1D profiles
+        p1 = eqt.profiles_1d
+        if length(p1.psi) == length(r)
```

Comment:
If you put this `if` statement inside your `interpolate_1d_profile` function, you could reduce a lot of code. So something like
```
function interpolate_1d_profile(ori_1D::AbstractVector{<:Real}, N::Int)
    if length(ori_1D) == N
        return ori_1D
    else
        itp=IMASdd.interp1d(p1.psi_norm, ori_1D, :cubic)
        return itp.(range(0,1,N))
    end
end